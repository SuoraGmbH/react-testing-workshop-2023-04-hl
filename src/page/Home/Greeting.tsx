import React from "react";

interface Props {
  name?: string;
}

const Greeting: React.FunctionComponent<Props> = ({ name }) => {
  if (!name || name.trim() === '') {
    return <div>Hello stranger!</div>;
  }

  return <div>Hello, dear {name}!</div>;
};

export default Greeting;
