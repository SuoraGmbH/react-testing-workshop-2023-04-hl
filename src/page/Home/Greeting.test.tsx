import { render, screen } from "@testing-library/react";
import Greeting from "./Greeting";

it("renders without crashing", () => {
  render(<Greeting />);
});

it('displays some greeting', () => {
  render(<Greeting />)

  expect(screen.getByText('hello', {exact: false})).toBeVisible()
})

it('displays the given name', () => {
  render(<Greeting name="Fabian" />)

  expect(screen.getByText('Fabian', {exact: false})).toBeVisible()
})

it('displays Hello if given a name', () => {
  render(<Greeting name="Fabian" />)

  expect(screen.getByText('Hello', {exact: false})).toBeVisible()
})

it('renders according to snapshot', () => {
  const {container} = render(<Greeting name="Fabian" />)

  expect(container).toMatchSnapshot()
})

it('renders according to snapshot for unknown users', () => {
  const {container} = render(<Greeting />)

  expect(container).toMatchSnapshot()
})


//
// it("greets strangers nicely", () => {
//   render(<Greeting />);
//
//   expect(screen.getByText("Hello stranger!")).toBeVisible();
// });
//
// test("displays the given name", () => {
//   render(<Greeting name="kek" />);
//
//   expect(screen.getByText("Hello, dear kek!")).toBeVisible();
// });
//
// test("it greets persons with just spaces in their name as strangers", () => {
//   render(<Greeting name="      " />);
//
//   expect(screen.getByText("Hello stranger!")).toBeVisible();
// });
//
//
