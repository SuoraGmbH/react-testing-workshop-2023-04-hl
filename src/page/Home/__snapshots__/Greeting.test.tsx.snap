// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders according to snapshot 1`] = `
<div>
  <div>
    Hello, dear 
    Fabian
    !
  </div>
</div>
`;

exports[`renders according to snapshot for unknown users 1`] = `
<div>
  <div>
    Hello, stranger!
  </div>
</div>
`;
